### DISCUSSION TITLE
Remove TOTP (Authenticator app) after enabling WebAuthn

### DISCUSSION THREAD

**Discussion by savchenko** [community/community#54621] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Question

### Body

I am unable to remove old TOTPs linked to the account after enabling WebAuthn.

Is this a bug or a design feature?

**Comment by hrishikesh-kocharekar-dev** [comment]
It's not clear which specific platform or service you are referring to, but generally speaking, the ability to remove old TOTPs (Time-Based One-Time Passwords) after enabling WebAuthn (Web Authentication) depends on the design and implementation of the platform or service.

WebAuthn is a relatively new authentication standard that allows users to authenticate with a device or token that is bound to their user account, rather than using passwords or TOTPs. If a platform or service supports WebAuthn, they may choose to allow users to remove old TOTP credentials after enabling WebAuthn. However, some platforms or services may not provide this feature, or may have specific requirements or restrictions for removing old TOTPs.

If you are having difficulty removing old TOTP credentials after enabling WebAuthn, I would recommend checking the documentation or support resources for the specific platform or service you are using. You may also want to contact their support team for further assistance.

**Reply by savchenko** [reply]
I am referring to (obviously) GitHub itself.

**Comment by iamreiyn** [comment, marked as answer]
Hello, @savchenko 

Unfortunately, as of now, there is no way to remove TOTP (Authenticator app) directly. However, **you can remove it by signing up for the 2FA again using the SMS/Text message method**. So, either you must have the Authenticator app method or Text/SMS method configured.

In order to do that:

1. Open your GitHub **Settings**

![procedure](https://i.postimg.cc/mkjzYZwF/Screenshot-20230508-051329-Edge.jpg)

2. Navigate to **Password and authentication**

![procedure](https://i.postimg.cc/9XY751hF/Smart-Select-20230508-052308-Edge.jpg)

3. Click on the **Enabled** button next to "Two factor authentication"

After it is disabled, you can enable it again using the SMS/text option and then reconfigure your WebAuthn 

Thank you!

**Reply by jbclements** [reply]
Erm. Shame about this. Seems it wouldn't be that hard to fix?

**Reply by warasilapm** [reply]
Yeah this is pretty terrible, especially if you're a part of organizations which require 2FA. What if an authenticator token is compromised? What are we supposed to do?
